AGENTS.md

```
@@ -3,3 +3,195 @@
+### Layering Validation
+```bash
+npm run valid-layers-check
+```
+
+## Writing and Testing Extensions
+
+VS Code's built-in extensions live in the `extensions/` folder. This is "Inception-level" development—you're building extensions that ship with the editor you're also developing.
+
+### Extension Structure
+Each extension in `extensions/` follows standard VS Code extension conventions:
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)